WordPress/src/main/java/org/wordpress/android/ui/comments/SharedCommentDetailFragment.kt

```diff
@@ -35,17 +87,118 @@ abstract class SharedCommentDetailFragment:CommentDetailFragment() {
+                site,
+                comment,
+                CommentStatus.APPROVED
+            )
+        }
         textMoreOptions.setOnClickListener { showModerationBottomSheet() }
     }
 
+    private fun CommentApprovedBinding.bindApprovedView() {
+        root.isVisible = true
+        meGravatarLoader.load(
+            newAvatarUploaded = false,
+            avatarUrl = meGravatarLoader.constructGravatarUrl(accountStore.account.avatarUrl),
+            imageView = replyAvatar,
+            imageType = ImageType.USER,
+            injectFilePath = null,
+        )
+
+        if (comment.iLike) {
+            handleLikeCommentView(
+                textLikeComment,
+                R.color.inline_action_filled,
+                R.drawable.star_filled,
+                R.string.comment_liked
+            )
+        } else {
+            handleLikeCommentView(
+                textLikeComment,
+                R.color.menu_more,
+                R.drawable.star_empty,
+                R.string.like_comment
+            )
+        }
+
+        textLikeComment.setOnClickListener {
+            viewModel.likeComment(comment, site)
+            sendLikeCommentEvent(comment.iLike.not())
+        }
+        cardReply.setOnClickListener { }
+        textReply.text = getString(R.string.comment_reply_to_user, comment.authorName)
+    }
+
+    /**
+     * Handle like comment text based on the liked status
+     */
+    private fun handleLikeCommentView(textView: TextView, colorId: Int, drawableId: Int, stringId: Int) {
+        textView.text = getString(stringId)
+        val color = ContextCompat.getColor(textView.context, colorId)
+        textView.setTextColor(color)
+        ContextCompat.getDrawable(textView.context, drawableId)
+            ?.apply { setTint(color) }
+            ?.let { textView.setCompoundDrawablesWithIntrinsicBounds(it, null, null, null) }
+    }
+
+    protected open fun sendLikeCommentEvent(liked: Boolean) {
+        AnalyticsUtils.trackCommentActionWithSiteDetails(
+            if (liked) AnalyticsTracker.Stat.COMMENT_LIKED else AnalyticsTracker.Stat.COMMENT_UNLIKED,
+            mCommentSource.toAnalyticsCommentActionSource(),
+            site
+        )
+    }
+
     private fun showModerationBottomSheet() {
-        ModerationBottomSheetDialogFragment.newInstance()
-            .show(childFragmentManager, ModerationBottomSheetDialogFragment.TAG)
+        ModerationBottomSheetDialogFragment.newInstance(
+            ModerationBottomSheetDialogFragment.CommentState(
+                canModerate = enabledActions.canModerate(),
+                canMarkAsSpam = enabledActions.canMarkAsSpam(),
+                canTrash = enabledActions.canTrash(),
+            )
+        ).apply {
+            onApprovedClicked = { viewModel.dispatchModerationAction(site, comment, CommentStatus.APPROVED) }
+        }.show(childFragmentManager, ModerationBottomSheetDialogFragment.TAG)
     }
 }
+
+object CommentExtension {
```

Comment:
Q: Why did you choose to define the extensions inside an object class?
I'm not suggesting a change here but I usually define extensions on at the package level on a separate file and want to understand the advantage of each approach 🙇